sdk/formrecognizer/ai-form-recognizer/MIGRATION-v3_v4.md

```diff
@@ -0,0 +1,232 @@
+| ------------------------- | ------------------------- | ----------- | ----------------------------- |
+| `FormRecognizerClient` | `DocumentAnalysisClient` | Class | This class replaces the former and has no methods in common with it. See the Section on `DocumentAnalysisClient` below. |
+| `FormTrainingClient` | `DocumentModelAdministrationClient` | Class | This class replaces the former, and renames/reworks several of its methods. See the section on `DocumentModelAdministrationClient` below. |
+| `RecognizedForm` and `RecognizedFormArray` | `AnalyzedDocument` | Interface | The `pages` property was removed. Pages are now accessed through `AnalyzeResult`. The properties `formType` and `formTypeConfidence` are now `docType` and `confidence` respectively. |
+| `FormField` and variants (e.g. `FormArrayField`, `FormStringField`, etc.) | `DocumentField` | Tagged (Branded) Union | The tag (or discriminator) has been renamed to `kind` from `valueType`. The `DocumentArrayField` and `DocumentObjectField` variants have their values in properties named `values` and `properties` respectively (previously, both used `value`). `name`, `labelData`, and `valueData` were removed. |
+| `FormPage` and `FormPageArray`                                            | `DocumentPage`                      | Interface              | The `tables` property was removed. Tables are now accessed through the `AnalyzeResult`. `textAngle` was renamed `angle`. There is now a `words` property in addition to `lines` that contains the words on the page. |
+| `FormLine` | `DocumentLine` | Interface | `appearance`, `kind`, and `words` were removed. `styles` is now a top-level property of the `AnalyzeResult`, and it replaces the `appearance` property of this type. Additionally, `words` are now properties of the `DocumentPage` and can be related to a `DocumentLine` through the line's `spans`. The `text` property was renamed `content`. |
+| `FormWord` | `DocumentWord` | Interface | `pageNumber` and `kind` were removed. `text` was renamed to `content`. `boundingBox` was replaced with `boundingRegions` (see the section on `boundingRegions` below). |
+| `FormSelectionMark` | `DocumentSelectionMark` | Interface | Removed `kind`. `boundingBox` was replaced with `boundingRegions` (see the section on `boundingRegions` below). |
+| `TextStyle` | `DocumentStyle` | Interface | Renamed `styleConfidence` to `confidence`, and replaced `styleName` with a simple boolean named `isHandwritten`. |
+| `FormTable` and `FormTableCell` | `DocumentTable` and `DocumentTableCell` | Interface | The `pageNumber` field was removed from `DocumentTable`. `isHeader`, `isFooter`, `fieldElements`, and `pageNumber` were removed from `DocumentTableCell`. The `text` field was renamed to `content`. Both types have replaced their `boundingBox` property with `boundingRegions`. (See the section on `boundingRegions` below.) |
+| `` | `` |  |  |
+| `` | `` |  |  |
+| `` | `` |  |  |
+| `` | `` |  |  |
+| `` | `` |  |  |
```

Comment:
awesome way to display the info! btw were the blank lines at the end left intentionally?

Reply:
Yes they're for me to copy/paste while the document is WIP :slightly_smiling_face: